### PR TITLE
Fix issues with building codebuild projects on startup

### PR DESCRIPTION
**Pull request by davidmoore1**

### CODE REVIEW

File: application/console/components/AwsStartupAction.php [hunk 1]
```diff
@@ -62,6 +73,26 @@ public function performAction()
...
         }
     }
+    private function copyDefaultProjectFolder($logger)
+    {
+        try {
+            $prefix = Utils::getPrefix();
+            echo "[$prefix] AwsStartupAction: copyDefaultProjectFolder" . PHP_EOL;
+            $s3 = new S3();
+            $currentDir = getcwd();
+            $sourceFolder = '/data/console/views/cron/scripts/project_default';
```
Comment (chrisvire) [inline application/console/components/AwsStartupAction.php]: I see you got `getcwd` but used a full path that would only work in docker.  Did you intent to build a path?

File: application/console/components/AwsStartupAction.php [hunk 2]
```diff
@@ -7,11 +7,13 @@
 use common\components\Appbuilder_logger;
 use common\components\CodeBuild;
 use common\components\IAmWrapper;
+use common\components\S3;
 
 use common\helpers\Utils;
 
 class AwsStartupAction extends ActionCommon
 {
+    private $s3;
```
Comment (chrisvire) [inline application/console/components/AwsStartupAction.php]: Why is this a private member when it is only used in copyDefaultProjectFolder?

File: application/console/components/AwsStartupAction.php [hunk 3]
```diff
@@ -38,16 +41,24 @@ public function performAction()
...
 
-           // Build publish role if necessary
+            // Build publish role if necessary
+            // Copy default file
             $projectName = 'publish_app';
+            $prefix = 'arn:aws:s3:::';
+            $bucket = S3::getArtifactsBucket();
+            $baseUrl = 'default/default.zip';
+            $defaultLocation = $prefix . $bucket . "/" . $baseUrl;
+            $defaultLocation = str_replace('"', "", $defaultLocation);
```
Comment (chrisvire) [inline application/console/components/AwsStartupAction.php]: Above you did this style of str_replace on just the `S3::getArtifactsBucket()` and then here again against the whole path.  Is the problem that sometimes the result of `S3::getArtifactsBucket()` includes quotes in the value.  Will this cause problems in other situations?  Should it be fixed in `S3::getArtifactsBucket()` and possibly other S3 methods?

File: application/console/components/AwsStartupAction.php (hunk 1 above)
Comment (davidmoore1) [reply application/console/components/AwsStartupAction.php]: I copied that from the other pr, saw it was not used and still forgot to take it out.  I will correct that

File: application/console/components/AwsStartupAction.php (hunk 2 above)
Comment (davidmoore1) [reply application/console/components/AwsStartupAction.php]: I'll fix this.  I think I was leaving it thinking towards merging it but I'll change it

Comment (davidmoore1) [reply application/console/components/AwsStartupAction.php]: I'll fix this.  I think I was leaving it thinking towards merging it but I'll change it

**Pull request merged**